libcontainer/container_setup.go

```diff
@@ -0,0 +1,150 @@
+package libcontainer
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/opencontainers/runc/libcontainer/configs"
+	"github.com/opencontainers/runc/libcontainer/system"
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink/nl"
+	"golang.org/x/sys/execabs"
+	"golang.org/x/sys/unix"
+)
+
+// NsExecSyncMsg is used for communication between the parent and child during
+// container setup.
+type NsExecSyncMsg uint32
+
+const (
+	syncUsermapPls NsExecSyncMsg = iota + 0x40
+	syncUsermapAck
+	syncRecvPidPls
+	syncRecvPidAck
+	syncTimeOffsetsPls
+	syncTimeOffsetsAck
+)
+
+type NsExecSetup struct {
+	process *containerProcess
+}
+
+const bufSize int = 4
```

Comment:
nit: I think `int` here is redundant, you can omit it as it will default to `int` anyway.